File: trunk/cgi-bin/gnuplot-scripts/plot_f_of_x.pl
#!/usr/bin/perl
use strict;
use warnings;

use lib "./gnuplot-scripts";
use Common;

my $mode = shift or die "no mode";
my $function = shift or die "no function";

my $png_settings = Common->new->gnuplot_png_settings();
my $math_settings = Common->new->gnuplot_math_settings();
my $plot_cmd = get_plot_cmd($mode, $function);

print "
$png_settings
$math_settings
$plot_cmd
";

sub get_plot_cmd {
  my $mode = shift or die "no mode";
  my $function = shift or die "no function";

  return "plot $function" if $mode eq 'math-2d';
  return "splot $function" if $mode eq 'math-3d';
  die "bad mode: $mode";
}

File: trunk/cgi-bin/gnuplot-scripts/plot_f_of_x.hs
#!/usr/lib/ghc-6.6/bin/runghc
import System
import Common 

main = do args <- getArgs
          let usagemsg = "usage examples: $ runghc plot_f_of_x.hs math-2d 'sin (x)'\n" ++
                         "                $ runghc plot_f_of_x.hs math-3d 'sin (x)'"
          case args of
            [style,function] -> gen_gnuplot_script style function
            _                    -> error $ "arg count does not equal 2. args: : " ++ ( show args ) ++ "\n" ++ usagemsg

gen_gnuplot_script style function = case style of 
                                      "math-2d" -> do putStrLn $ gnuplot_png_settings ++ "\n" ++
                                                                 gnuplot_math_settings ++ "\n" ++
                                                                 "plot " ++ function
                                      _         -> error $ "bad stle: " ++ style
